components/Account/Memberships/Claim.js

```diff
@@ -69,20 +65,39 @@ class ClaimMembership extends Component {
     }))
   }
   handleVoucherCode (value, shouldValidate, t) {
+    const sanitizedValue = this.sanitizeVoucherCode(value)
+
     this.setState(FieldSet.utils.mergeField({
       field: 'voucherCode',
       value,
-      error: (
-        value.trim().length <= 0 && t('memberships/claim/voucherCode/label/error/empty')
-      ),
+      error:
+        (
+          sanitizedValue.length === 0 &&
+          t('memberships/claim/voucherCode/label/error/empty')
+        ) ||
+        (
+          !this.isMembershipVoucherCode(sanitizedValue) &&
+          !this.isAccessGrantVoucherCode(sanitizedValue) &&
+          t('memberships/claim/voucherCode/label/error/unrecognized')
+        ),
       dirty: shouldValidate
     }))
   }
+  isMembershipVoucherCode (voucherCode) {
+    return voucherCode.length === 6
+  }
+  isAccessGrantVoucherCode (voucherCode) {
+    return voucherCode.length === 5
+  }
+  sanitizeVoucherCode (value) {
```

Comment:
this method, like the two above, does not need to be on the class. define this one before, export and use in `pages/claim.js`:
```js
export const sanitizeVoucherCode = value => value
  .replace(/[^a-zA-Z0-9]/g, '')
  .trim()
  .substr(0, 6) // or 10 but currently we only have up to 6 right?
```